src/Symfony/Component/Validator/ValidatorBuilder.php

```diff
@@ -187,31 +188,56 @@ public function addMethodMappings(array $methodNames): static
     }
 
     /**
-     * Enables annotation and attribute based constraint mapping.
+     * @deprecated since Symfony 6.4, use "enableAttributeMapping()" instead.
      *
      * @return $this
      */
     public function enableAnnotationMapping(): static
+    {
+        trigger_deprecation('symfony/validator', '6.4', 'Method "%s()" is deprecated, use "enableAttributeMapping()" instead.', __METHOD__);
+
+        return $this->enableAttributeMapping();
+    }
+
+    /**
+     * Enables attribute-based constraint mapping.
+     *
+     * @return $this
+     */
+    public function enableAttributeMapping(): static
     {
         if (null !== $this->metadataFactory) {
-            throw new ValidatorException('You cannot enable annotation mapping after setting a custom metadata factory. Configure your metadata factory instead.');
+            throw new ValidatorException('You cannot enable attribute mapping after setting a custom metadata factory. Configure your metadata factory instead.');
         }
 
-        $this->enableAnnotationMapping = true;
+        $this->enableAttributeMapping = true;
 
         return $this;
     }
 
     /**
-     * Disables annotation and attribute based constraint mapping.
+     * @deprecated since Symfony 6.4, use "disableAttributeMapping()" instead
      *
      * @return $this
      */
     public function disableAnnotationMapping(): static
     {
-        $this->enableAnnotationMapping = false;
+        trigger_deprecation('symfony/validator', '6.4', 'Method "%s()" is deprecated, use "disableAttributeMapping()" instead.', __METHOD__);
+
         $this->annotationReader = null;
```

Comment:
Let's move this line to `disableAttributeMapping()` as well. The two methods should expose the same behavior to ease the migration.